lib/event-target.js

```diff
@@ -180,7 +186,14 @@ EventTarget.prototype = Object.create(
                 // This doesn't process capturing phase and bubbling phase.
                 // This isn't participating in a tree.
                 while (node != null) {
-                    node.listener.call(this, wrapped);
+                    var isObjListener = hasHandleEventMethod(node.listener);
+                    if (isObjListener) {
+                        node.listener.handleEvent.bind(node.listener).call(this, wrapped);
```

Comment:
I tried removing it, the first test I added then fails since `this` is not correct.